Rubberduck.Parsing/VBA/RubberduckParserState.cs

```diff
@@ -271,6 +271,13 @@ public void RefreshProjects(IVBE vbe)
             }
         }
 
+        private List<SyntaxErrorEventArgs> _errors = new List<SyntaxErrorEventArgs>();
```

Comment:
I think a `List is not a good choice here, since we parse on multiple threads. So, elements can be added concurrently. 
How about a `ConcurrentDictionary<QualifiedModuleName, ConcurrentBag<SyntaxErrorEventArgs>>` or if we decide the collect the errors first for each parse, a `ConcurrentDictionary<QualifiedModuleName, List<SyntaxErrorEventArgs>>?